tests/test_batch_edit.py

```
@@ -312,9 +318,81 @@ def test_batch_insert_anchor_not_found(self, multi_para_doc):
+        ]
+        with pytest.raises(BatchOperationError) as exc:
+            doc.batch_edit(ops)
+        err = exc.value
+        assert err.operation_index == 1
+        assert "Unknown action" in err.reason
+
+    def test_non_batch_value_error_unchanged(self):
+        """Malformed paragraph refs outside batch paths still raise plain ValueError."""
+        from docx_editor import ParagraphRef
+
+        with pytest.raises(ValueError):
```

Ruff (0.15.10)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)